src/main/java/org/openrewrite/kotlin/Assertions.java

```diff
@@ -196,6 +209,36 @@ public static UncheckedConsumer<SourceSpec<?>> spaceConscious() {
         };
     }
 
+    private static void assertValidTypes(TypeValidation typeValidation, J sf) {
```

Comment:
@knutwannheden @kunli2, I don't think mixins will help us customize the `FindMissingTypes` recipe. Initially, I thought about creating a `FindMissingKotlinTypes` recipe/visitor, which could arguably be helpful. But I recall Jonathan preferring nearly identical recipes not to be created. Eventually, if we find there is a need for the augmented recipe, we can move the code.

What do you think? Is it fine to have the visitor in Assertions for now or would you prefer a recipe?

Reply:
At some point I would be interested in the details regarding the mixins, do that we can try to come up with a fix or another approach for this. An alternative might be delegation kind of like we do in the printer.

For now I think what you currently have is good. Having type validations working is really a step forward. Thanks for fixing this!